### PR TITLE
Simply use ruamel.yaml for YAML parsing

### PR DESCRIPTION
**Pull request by jakirkham**
Rewrite logic that was using PyYAML to use ruamel.yaml as is used elsewhere in the code. This eliminates the need for PyYAML as a dependency.

<hr>

Checklist
* [x] Pushed the branch to main repo
* [x] CI passed on the branch

### CODE REVIEW

**Comment by jakirkham** [comment]
@beckermr could you also please review this PR?

File: conda_forge_webservices/commands.py [hunk 1]
```diff
@@ -643,19 +643,23 @@ def add_user(repo, user):
 
     recipe_path = os.path.join(repo.working_dir, "recipe", "meta.yaml")
     co_path = os.path.join(repo.working_dir, ".github", "CODEOWNERS")
+    yaml = YAML()
```
Comment (beckermr) [inline conda_forge_webservices/commands.py:646]:
```suggestion
    yaml = YAML(typ="safe")
```

We should be careful since the recipe is from users.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: conda_forge_webservices/commands.py (hunk 1 above)
Comment (jakirkham) [reply conda_forge_webservices/commands.py:646]: Sounds reasonable

We do have a bunch of places where this is not set now. Maybe it's worth filing an issue to fix those up too (not sure if those will be as simple without looking more deeply)?

Comment (beckermr) [reply conda_forge_webservices/commands.py]: https://github.com/conda-forge/conda-forge-webservices/issues/600

**Pull request merged**